src/main/java/spoilagesystem/timestamp/LocalTimeStampService.java

```diff
@@ -133,15 +133,19 @@ public OffsetDateTime getTimeStamp(ItemStack item) {
 
     @Nullable
     private OffsetDateTime parseExpiryFromPersistentData(ItemMeta meta) {
-        String expiryString = meta.getPersistentDataContainer().get(expiryKey, STRING);
-        if (expiryString != null) {
-            try {
-                return OffsetDateTime.parse(expiryString, ISO_OFFSET_DATE);
-            } catch (DateTimeParseException exception) {
-                plugin.getLogger().log(SEVERE, "Failed to parse expiry from persistent data container", exception);
+        try {
+            String expiryString = meta.getPersistentDataContainer().get(expiryKey, STRING);
+            if (expiryString != null) {
+                try {
+                    return OffsetDateTime.parse(expiryString, ISO_OFFSET_DATE);
+                } catch (DateTimeParseException exception) {
+                    plugin.getLogger().log(SEVERE, "Failed to parse expiry from persistent data container", exception);
+                }
             }
+            return null;
+        } catch(Exception e) {
```

Comment:
As a general practice avoid catching generic exceptions. Using more specific types ensures we are handling failure cases gracefully, as a first-class citizen, and anything we don't know how to handle is made visible.

Reply:
Sounds good, thanks for the tip.